docs/protocol.md

```diff
@@ -662,7 +673,233 @@ Response (in case of success)
+
+<details>
+<summary>
+💡 See the full specification
+</summary>
+  
+```json5
+// ../protocol/query/method/heartbeat.json
+
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/popstellar/master/protocol/query/method/heartbeat.json",
+    "description": "Heartbeat message",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "method": {
+            "description": "[String] operation to be performed by the query",
+            "const": "heartbeat"
+        },
+        "params": {
+            "description": "[Array] of objects containing each a channel and the list of ids from that channel",
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "maps": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                            "channel": {
+                                "description": "[String] name of the channel",
+                                "type": "string",
+                                "pattern": "^/root(/[^/]+)*$"
+                            },
+                            "ids": {
+                                "description": "[Array] of message_ids corresponding to that channel",
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            }
+                        },
+                        "required": ["channel", "ids"]
+                    }
+                }
+            },
+            "required": ["maps"]
+        },
+        "jsonrpc": {
+            "$comment": "Defined by the parent, but needed here for the validation"
+        },
+
+        "id": {
+            "type": "integer"
+        }
+    },
+    "required": ["method", "params", "id", "jsonrpc"]
+}
+
+```
+</details>
+
+## Retrieving messages from server using ids 
+
+🧭 **RPC Message** > **Query** > **getMessagesById**
   
+This call is generally triggered upon the reception of a hearbeat message, in case the server hasn't already delivered some of the messages contained in such heartbeat but it could also used in other contexts. The purpose is to request missed messages so far.
+
+RPC 
+
+```json5
+// ../protocol/examples/query/getMessagesById/getMessagesById.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 6,
+    "method": "getMessagesById",
+    "params": {
+        "maps": [
```

Comment:
Same comment as before:

Why is the name "maps"? Because it contains the messages per channel? The I'd name it something like that, i.e. "messages_by_id"?